audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,506 @@
+          let lastAmplitude = amplitudes[0];
+          for (let i = 1; i < amplitudes.length; i++) {
+            should(amplitudes[i], 'The attacking amplitude at iteration ' + i)
+                .beLessThan(amplitudes[i - 1],
+                            'the attacking amplitude at iteration ' + (i - 1));
+          }
+
+          // 4. Closed Phase: the signal must remain at 0 for a non-trivial
+          // period of time.
+          should(closedPhase, 'The closed phase').beConstantValueOf(0);
+          should(closedPhase.length, 'The closed phase')
+              .beGreaterThanOrEqualTo(100);
+
+          // 5. Releasing Phase: the noisegate signal increases until it matches
+          // the original signal.
+          let fromClosedToEndPhase =
+              noiseGateOutput.slice(firstReleaseIndex, noiseGateOutput.length);
+          let fromClosedToEndPhaseReference =
+              originalOutput.slice(firstReleaseIndex, noiseGateOutput.length);
+          let openIndex = _getFirstIndexBeforeRemainingSamplesMatch(
+              fromClosedToEndPhase, fromClosedToEndPhaseReference);
+          let gainIncreasingPhase = fromClosedToEndPhase.slice(0, openIndex);
+          let openPhase2 =
+              fromClosedToEndPhase.slice(openIndex, firstReleaseIndex);
+          let openPhase2Reference =
+              fromClosedToEndPhaseReference.slice(openIndex, firstReleaseIndex);
+
+          should(gainIncreasingPhase.length, 'The number of steps when the signal is increasing').beGreaterThanOrEqualTo(attackAndReleaseSteps, 'the number of programmed attack and release steps.');
+
+          amplitudes = _getAveragedAmplitudes(gainIncreasingPhase, 10);
+          lastAmplitude = amplitudes[0];
+          for (let i = 1; i < amplitudes.length; i++) {
+            should(amplitudes[i], 'The attacking amplitude at iteration ' + i).beGreaterThan(amplitudes[i-1], 'the attacking amplitude at iteration ' + (i-1));
+          }
+
+          // 6. Open Phase 2: The gate eventually re-opens.
+          should(openPhase2.length, 'The second open phase').beGreaterThanOrEqualTo(100);
+          should(openPhase2, 'The second open phase of noise gate output')
+              .beEqualToArray(
+                  openPhase2Reference,
+                  'the original samples generated by the oscillator');
+
+          }).then(() => task.done());
+      }) 
+      
+      
+      // Verify that the noise gate computes an envelope for an audio signal
+      // that is on the same scale as web audio's gain node. If the gain of a
+      // signal is set to be beneath the threshold, then the signal should be
+      // muted. The gain of an oscillator is gradually decreased to some 
+      // threshold T at timeToThreshold, and the output of the noise gate 
+      // should be 0 after timeToThreshold and unaltered before timeToThreshold
+      // within an acceptable margin of error.
+      audit.define('envelope-and-threshold', (task, should) => {
```

Comment:
What does "optimize alpha" mean?  What makes it optimal?  What is the criteria for being optimal?